### PR TITLE
[Filestream] New Input package

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

This PR adds a custom input package for the filestream input, similar to how we have input packages for other similar custom inputs.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by P1llus** [comment]
While the package is now in a working state, there is still some work that needs to be done on the documentation side, especially:

1. The README.md is currently just the same as from the filebeat documentation, do we want to update it or keep it?
2. The description for UI options in manifest.yml can be quite long, they are also based on the documentation we already have for those fields.
3. We need to make sure all the UI fields have the correct type (bool or text).

**Comment by P1llus** [comment]
Working with @kvch to see what we can do on the documentation front. Maybe we could minimize the documentation in the UI, and link more to the documentation page?

File: packages/filestream/data_stream/generic/agent/stream/filestream.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+backoff.max: {{backoff_max}}
+{{/if}}
+{{#if file_identity_native}}
+file_identity.native: {{file_identity_native}}
+{{/if}}
+{{#if file_identity_path}}
+file_identity.path: {{file_identity_path}}
+{{/if}}
+{{#if file_identity_inode_marker_path}}
+file_identity.path: {{file_identity_inode_marker_path}}
```
Comment (kvch) [inline packages/filestream/data_stream/generic/agent/stream/filestream.yml.hbs:74]:
```suggestion
file_identity.inode_marker.path: {{file_identity_inode_marker_path}}
```

File: packages/filestream/data_stream/generic/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        show_user: false
+      - name: ignore_inactive
+        type: text
+        title: Ignore Inactive
+        description: |
+          If this option is enabled, Elastic Agent ignores every file that has not been updated since the selected time. Possible options are since_first_start and since_last_start.
+        required: false
+        show_user: false
+      - name: close_on_state_changed_inactive
+        type: bool
```
Comment (kvch) [inline packages/filestream/data_stream/generic/manifest.yml:135]: It is not a boolean option. It expects a time interval, e.g. 5m.

**Review by kvch — APPROVED** [review]

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -55,6 +55,7 @@
 /packages/elasticsearch @elastic/infra-monitoring-ui
 /packages/etcd @elastic/integrations
 /packages/f5 @elastic/security-external-integrations
+/packages/filestream @elastic/obs-service-integrations
```
Comment (jsoriano) [inline .github/CODEOWNERS:58]: Should this one be owned by @elastic/elastic-agent-data-plane? Related discussion in https://github.com/elastic/integrations/pull/3202.

**Pull request merged**